### ISSUE TITLE
Feature Array to Object

### ISSUE THREAD

**Issue by T00mm** [excaliburjs/excalibur-tiled#23]
https://github.com/excaliburjs/excalibur-tiled/blob/d9f002c620710ef527e86ec502b1a1f621d15155/src/ITiledMap.ts#L10

At the moment it is hard to easily get the layer you want, you mist iterate and check names. It would be much easier if we could use and object so we can access layers through their name

Current code needs:

```
  var gameObjectsLayer;
  for(var layer of map.data.layers){
    if(layer.name === "GameObjects"){
      gameObjectsLayer = layer;
      break;
    }
  }
```
Would be much nicer to use `map.data.layers["GameObjects"]`

What do you guys think?

**Comment by T00mm** [comment]
Same for "objects" in a layer. With Tiled we can name objects but to access them we must write clunky code atm.

**Comment by eonarheim** [comment]
@T00mm Agreed, the way it is now cumbersome. I generally like being able to get at things by name.

@kamranayub any objections?

**Comment by kevin192291** [comment]
Hey Guys! I was reading through this, and I had pulled a version of the tiled loader a long time ago and made this update on my local branch.
The main reason I made the change was for more efficient searches for when I was adding the ability to get tile properties.
I am wondering if this issue is still needing help. I would love to get what I have done merged into the repo if not. I am relying heavily on the tile properties right now.
Or am I just way out of date and this feature has already been added and I am just being annoying?

**Comment by eonarheim** [comment]
@kevin192291 For sure this does need some more help, I think we left off wanting a few things:
* An alternate more convenient api
* Strong typing
* Avoiding a breaking change
